src/test/java/co/rsk/bitcoinj/core/StoredBlockTest.java

```diff
@@ -97,10 +128,93 @@ public void serializeAndDeserializeCompact_forMax12bytesChainWork_works() {
+        StoredBlock blockToStore = new StoredBlock(block, TOO_LARGE_WORK_V1, blockHeight);
+
+        // serialize block should throw illegal argument exception
+        blockToStore.serializeCompact(blockBuffer);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void serializeCompact_for32bytesChainWork_throwsException() {
+        StoredBlock blockToStore = new StoredBlock(block, MAX_WORK_V2, blockHeight);
 
         // serialize block should throw illegal argument exception
         blockToStore.serializeCompact(blockBuffer);
     }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void serializeCompact_forMoreThan32bytesChainWork_throwsException() {
+        StoredBlock blockToStore = new StoredBlock(block, TOO_LARGE_WORK_V2, blockHeight);
+
+        // serialize block should throw illegal argument exception
+        blockToStore.serializeCompact(blockBuffer);
+    }
+
+
+    @Test(expected = IllegalArgumentException.class)
+    public void serializeCompactV2_forNegativeChainWork_throwsException() {
+        StoredBlock blockToStore = new StoredBlock(block, NEGATIVE_CHAIN_WORK, blockHeight);
+
+        // serialize block should throw illegal argument exception
+        blockToStore.serializeCompactV2(blockBuffer);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void serializeCompactV2_forMoreThan32bytesChainWork_throwsException() {
+        StoredBlock blockToStore = new StoredBlock(block, TOO_LARGE_WORK_V2, blockHeight);
+
+        // serialize block should throw illegal argument exception
+        blockToStore.serializeCompactV2(blockBuffer);
+    }
+
+    @Test
+    public void serializeCompactV2_forZeroChainWork_works() {
+        testSerializeCompactV2(ZERO_CHAIN_WORK);
+    }
+
+    @Test
+    public void serializeCompactV2_forSmallChainWork_works() {
+        testSerializeCompactV2(SMALL_CHAIN_WORK);
+    }
+
+    @Test
+    public void serializeCompactV2_for8BytesChainWork_works() {
+        testSerializeCompactV2(EIGHT_BYTES_WORK_V1);
+    }
+
+    @Test
+    public void serializeCompactV2_for12ByteChainWork_works() {
+        testSerializeCompactV2(MAX_WORK_V1);
+    }
+
+    @Test
+    public void serializeCompactV2_forTooLargeWorkV1_works() {
+        testSerializeCompactV2(TOO_LARGE_WORK_V1);
+    }
+
+    @Test
+    public void serializeCompactV2_for32BytesChainWork_works() {
+        testSerializeCompactV2(MAX_WORK_V2);
+    }
+
+    private void testSerializeCompactV2(BigInteger chainWork) {
+        StoredBlock blockToStore = new StoredBlock(block, chainWork, 0);
+        ByteBuffer buf = ByteBuffer.allocate(StoredBlock.COMPACT_SERIALIZED_SIZE_V2);
+        blockToStore.serializeCompactV2(buf);
+        // assert serialized size and that the buffer is full
+        assertEquals(StoredBlock.COMPACT_SERIALIZED_SIZE_V2, buf.position());
+    }
+
+    @Test
+    public void moreWorkThan_whenLargerVsSmallerChainWork_shouldReturnTrue() {
```

Comment:
Should we include this test?

Reply:
I saw it in the implementation of the bitcoinj team and I wanted to added as well. But it is unrelated to the `serializeCompactV2` method. I'll remove it to avoid confusion.